urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/calculator/CalculatorHandler.java

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2009 Hao Xu
+ertranne@hotmail.com
+
+This file is part of EriLex.
+
+EriLex is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+EriLex is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.calculator;
+
+import java.math.BigInteger;
+import java.util.Stack;
+import erilex.Handler;
+import erilex.Production;
+import erilex.data.CharStream;
+
+/**
+ *
+ * @author ertri
+ */
+public class CalculatorHandler implements Handler {
+
+    private Stack<BigInteger> stack = new Stack();
+    private Stack<String> opStack = new Stack();
+    public BigInteger res = null;
+    private String op = null;
+
+    public void handle(Production production, CharStream cs, CharStream.Label start, CharStream.Label finish) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.